### DISCUSSION TITLE
Vercel is not using Node 18.12.1 (Node 18.x)

### DISCUSSION THREAD

**Discussion by saguilarolmo** [vercel/community#4431] [Bug Report]
### Page to Investigate

https://vercel.com/changelog/node-js-18-lts-is-now-available

### Steps to Reproduce

I have an app in Next that I deploy in Vercel with the Node 18.x configuration.
In my application, I define a POST method that returns the response using the following code:
`return NextResponse.json(response.content, {status:200})`

After some research, we found out that  NextResponse.json does not work with Node version 18.0.0. This method works from version 18.3.0 onwards.

When the code is deployed in Vercel and the app is executed using the url obtained from Vercel, the error characteristic of this problem appears.

To solve it, I have to add in my package.json the following code:
`"engines": {
    "node": "18.12.1"
  }`

This makes me suspect that Vercel is not using Node version 18.12.1, as it says in the doc.

Also, if I look at the Vercel logs I can see this in the deployment process: 

`
> Detected Turbo. Adjusting default settings...
--
12:04:00.427 | Running "install" command: `npm install --prefix=../..`...
12:04:01.685 | npm WARN EBADENGINE Unsupported engine {
12:04:01.686 | npm WARN EBADENGINE   package: 'smart-insights@1.0.0',
12:04:01.686 | npm WARN EBADENGINE   required: { node: '18.12.1' },
12:04:01.686 | npm WARN EBADENGINE   current: { node: 'v18.17.1', npm: '9.6.7' }
12:04:01.686 | npm WARN EBADENGINE }
`
So, I'm not sure what version of Node is being used in the end.

**Comment by dudeOFprosper** [comment]
Thank you for reporting this @saguilarolmo! I could be wrong but: 

- I was under the assumption they would be using the latest from Node 18, with updates to patches and minor versions. So the documentation at the time of writing is outdated. 
- BUT: 18.17.1 is not the latest either, as node 18 is 18.18.2. 

So I think their recommendation is to target only the major version in your engines.json file - "v18" and hopefully they upgrade to the latest of node 18 sooner than later.
